config/config.go

```diff
@@ -30,6 +34,9 @@ type EncryptConfig struct {
 type DecryptConfig struct {
 	// map holding 'privkeys', 'x509s', 'gpg-privatekeys'
 	Parameters map[string][][]byte
+
+	// Pkcs11Config holds module directories etc. for pkcs11 encryption and decryption
+	Pkcs11Config *pkcs11.Pkcs11Config
```

Comment:
This is related to a pcs11 config file that declares in which paths the pkcs11 shared libraries/modules can be found.Every pkcs11URI object needs to be augmented with the data from that file so that the module can be found.

Pkcs11Config is itself more or less an object representing the contents of that file. Its contents could be copied into the map, but I am not sure whether this is a cleaner solution...

Reply:
I'm just thinking of other configurations and extensibility. The map provides a stable config interface to integrate with various runtimes - and allowing implementation of custom keywrappers without needing to change the interface. If we want to follow this pattern for PKCS11, then we should do this for the other protocols as well. But i fear backward/forward compatibility management will be a bit complex.

I think it will be nice for the config interface to be stable and the versioning of protocols be done on the protocol `keywrap` level since its metadata is freeform.